FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/ContainerRegistry/JsonSchemaCollectionsProvider.cs

```diff
@@ -0,0 +1,94 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using DotLiquid;
+using EnsureThat;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Synapse.SchemaManagement.Exceptions;
+using Microsoft.Health.Fhir.TemplateManagement;
+using Microsoft.Health.Fhir.TemplateManagement.Exceptions;
+using Microsoft.Health.Fhir.TemplateManagement.Models;
+using Newtonsoft.Json.Schema;
+
+namespace Microsoft.Health.Fhir.Synapse.SchemaManagement.ContainerRegistry
+{
+    public class JsonSchemaCollectionProvider
```

Comment:
I feel this class works more like a implementation of some `ISchemaProvider`, if we are not using the JSchema directly, we can add the interface like `Dictionary<string, FhirSchemaNode> GetSchemaAsync()`, then we can have two different implementations, local schema provider and acr schema provider.